internal/agent/device/dependency/dependency.go

```diff
@@ -552,11 +552,15 @@ func (m *prefetchManager) cleanupStaleTasks(seenTargets map[string]struct{}) {
 }
 
 func (m *prefetchManager) Cleanup() {
-	m.log.Debug("Prefetch cleanup: canceling active task and draining queue")
-
 	m.mu.Lock()
 	defer m.mu.Unlock()
 
+	if len(m.tasks) == 0 {
+		return
+	}
+
+	m.log.Debugf("Prefetch cleanup: canceling %d active tasks", len(m.tasks))
```

Comment:
note: lets really read our logs in debug make sure things like this get cleaned up. AKA do not log if nothing happens